lisa/energy_model.py

```diff
@@ -842,6 +842,67 @@ def _simple_pd_root(target):
             cpu_pds.append(PowerDomain(cpu=cpu, idle_states=names))
         return PowerDomain(children=cpu_pds, idle_states=[])
 
+    @classmethod
+    def from_debugfsEM_target(cls, target,
+            directory='/sys/kernel/debug/energy_model'):
+        """
+        Create an EnergyModel by reading a target filesystem on a device with
+        the new Simplified Energy Model present in debugfs.
+
+        This uses the energy_model debugfs used usptream to expose the
+        performance domains, their frequencies and power costs. This feature is
+        upstream as of Linux 5.1. It is also available on Android 4.19 and
+        later.
+
+        Wrt. idle states - the EnergyModel constructed won't be aware of
+        any power data or topological dependencies for entering "cluster"
+        idle states since the simplified model has no such concept.
+
+        Initialises only Active States for CPUs and clears all other levels.
+
+        :param target: Devlib target object to read filesystem from. Must have
+                       cpufreq and cpuidle modules enabled.
+        :returns: Constructed EnergyModel object based on the parameters
+                  reported by the target.
+        """
+
+        if 'cpuidle' not in target.modules:
+            raise TargetStableError('Requires cpuidle devlib module. Please ensure '
```

Comment:
kernel coding style has a sensible exception for line breaking rule: user-visible messages should not be split, to allow grepping for it directly. I propose we use that in LISA as well.